app/src/main/java/com/example/epamandroid/MyService.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.epamandroid
+
+import android.app.IntentService
+import android.content.Intent
+import android.os.Build
+import android.support.annotation.RequiresApi
+import android.widget.Toast
+import java.time.LocalDateTime
+
+class MyService : IntentService("MyService") {
+
+    private val myIntent: Intent = Intent(Constants.CUSTOM_ACTION)
+
+    private val colorList: MutableList<String> = ArrayList()
+
+    private var serviceStatus: Boolean = false
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onHandleIntent(intent: Intent) {
+        serviceStatus = intent.getBooleanExtra(Constants.SERVICE_STATE, false)
+
+        do {
+            Thread.sleep(350)
+            when {
+                serviceStatus -> myIntent.putExtra(
+                    Constants.BROADCAST_MESSAGE,
+                    colorList[(Math.random() * 100 % (colorList.size - 1)).toInt()]
+                )
+                !serviceStatus -> myIntent.putExtra(
+                    Constants.BROADCAST_MESSAGE,
+                    "#FFFFFF"
+                )
+            }
+            sendBroadcast(myIntent)
+        } while (serviceStatus)
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        setBackgroundList()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+
+        colorList.clear()
+        serviceStatus = false
+    }
+
+    private fun setBackgroundList() {
+        colorList.add("#FFAB00")
```

Comment:
How about this

![image](https://user-images.githubusercontent.com/16294951/54493286-193f5a80-48df-11e9-879c-905b45a0129c.png)

Reply:
@egorikftp redid